### PR TITLE
add version to ModeDef

### PR DESCRIPTION
**Pull request by erwinmombay**
Closes https://github.com/ampproject/amphtml/issues/2874

### CODE REVIEW

**Comment by erwinmombay** [comment]
@cramforce PTAL

File: src/mode.js [hunk 1]
```diff
@@ -69,13 +80,19 @@ function getMode_() {
...
+  }
+
   return {
     localDev: isLocalDev,
     // Triggers validation
     development: developmentQuery['development'] == '1' || window.AMP_DEV_MODE,
     minified: process.env.NODE_ENV == 'production',
     test: window.AMP_TEST,
     log: developmentQuery['log'],
+    version: version,
```
Comment (cramforce) [inline src/mode.js]: Lets remove this. We version lock the child frame, so it can read this itself, and we need to be careful with the URL length.

Comment (erwinmombay) [inline src/mode.js]: got it, will rename fullVersion to just version

Comment (erwinmombay) [inline src/mode.js]: done.

File: src/mode.js [hunk 2]
```diff
@@ -114,3 +130,37 @@ function parseQueryString_(queryString) {
...
+  if (isLocalDev) {
+    return version;
+  }
+
+  // TODO(erwinm, #2917): Get full version from AMP_CONFIG if it exists.
+  if (win.AMP_CONFIG && win.AMP_CONFIG.fullVersion) {
+    return win.AMP_CONFIG.fullVersion;
+  }
+
+  const selector = 'script[src$="/v0.js"]';
```
Comment (cramforce) [inline src/mode.js]: When do we need this code path?

Comment (erwinmombay) [inline src/mode.js]: so right now we don't have `win.AMP_CONFIG.fullVersion` so this is to look for v0.js and read the version from the script tags src when the page is on the cache. I should implement #2917 by EOD. so you're right im not sure we need it. let me know what you think.

Comment (erwinmombay) [inline src/mode.js]: removed.

added `win.AMP_CONFIG.v` during build

**Comment by erwinmombay** [comment]
@cramforce PTAL

**Comment by cramforce** [comment]
LGTM

**Pull request merged**